ocrd_network/ocrd_network/models/job.py

```diff
@@ -13,7 +13,7 @@ class StateEnum(str, Enum):
     failed = 'FAILED'
 
 
-class JobInput(BaseModel):
+class PYJobInput(BaseModel):
```

Comment:
Why did you rename them? I find the old names already good. What does prefix `PY` mean? Why call it `DBProcessorJob` instead of `Job`?

Reply:
> What does prefix PY mean? 

@tdoan2010, have you missed [this](https://github.com/OCR-D/core/pull/974/commits/8f320044fd0e7d719892220722b9b1d49a46d718#diff-8c6aa4d80e67e1fdd9981f9838bc6664707eb0699573ce1c6ddf346a5419f5f5R1-R4)? 
The prefix `PY` is not as descriptive as the `DB` though and a better replacement may be needed.

> Why did you rename them? I find the old names already good. Why call it DBProcessorJob instead of Job?

You may find them good but they are rather not descriptive and in some cases even confusing from the perspective of the entire `ocrd` repository. Consider the following two examples:

1) Naming the workspace database model just as `Workspace` (as it was before) collides on a conceptual level with `ocrd.Workspace`. Having a `DB` prefix is a fast hint that this is something related to the database model. The readability of the code increases - even people who never checked the `ocrd` repository before can easily distinguish `Workspace` from `DBWorkspace` without any context around these classes. This is also more convenient for projects that use `core` as a library. 
2) `Job` versus `DBProcessorJob` follows a similar logic. In the reference WebAPI implementation there is the `WorkflowJobDB` model which should be also refactored to `DBWorkflowJob`. Then it is clearly specified what `Job` stands for. Similar arguments stand for `PYJobInput` and `PYJobOutput` pydantic models that are used on the Processing Server for input and output, respectively. 
The conversion from `PYJobInput` to `DBProcessorJob` and from `DBProcessorJob` to `PYJobOutput`.

Naming properly has more value than many lines of comments to explain what is what. Regardless of the model class naming in the source code, the database collections could still be renamed to something more convenient, e.g.:
```python

class DBProcessorJob(Document):
    ....
    class Settings:
        name = "processor_jobs"
```

I think @bertsky and @kba should also drop an opinion on that. 
In case the older naming convention is better for the OCR-D community than the current one, I will revert the naming.

Reply:
I'm against this new naming style. It makes the name unnecessary cryptic. Our classes are not a standalone piece of code, but with context. For example, class `Job` is located inside `models` package, therefore we know that it is a model.

For naming conflict, e.g. `Workspace` ocrd-object vs `Workspace` model, it is easily solved with fully qualified name, `ocrd.Workspace` and `ocrd_network.models.Workspace`.

Reply:
> It makes the name unnecessary cryptic.

I agree that `PY` in this case is cryptic and not good as I wrote in my first message.

> For example, class `Job` is located inside `models` package, therefore we know that it is a model.

This still does not resolve the different types of jobs - `ProcessingJob`, and `WorkflowJob`.

> For naming conflict, e.g. Workspace ocrd-object vs Workspace model, it is easily solved with fully qualified name, ocrd.Workspace and ocrd_network.models.Workspace.

How do you import `ocrd.Workspace` and `ocrd_network.models.Workspace` in the same context without having to use a placeholder for either of the two, e.g.:

```python
from ocrd.workspace import Workspace
from ocrd_network.models.database import Workspace as SOME_PLACEHOLDER
from ocrd_network.models.workspace import Workspace as SOME_OTHER_PLACEHOLDER
```

And any time these are needed in the same context, different placeholders will be introduced. Not to mention all the circular import problems I had to deal with in the reference WebAPI implementation in the past.

vs.

```python
from ocrd.workspace import Workspace
from ocrd_network.models.database import DBWorkspace
from ocrd_network.models.workspace import PYWorkspace
```

Where the placeholders are no longer needed.

Moreover, why should a user of library care know the exact path of imported things? As a user, I would rather do:
```python
from ocrd_network import DBWorkspace, PYWorkspace
```

than 

```python
from ocrd_network.models.database import Workspace as SOME_PLACEHOLDER
from ocrd_network.models.workspace import Workspace as SOME_OTHER_PLACEHOLDER
```

The second one is asking for troubles when the paths no longer match for whatever reason.

Reply:
I agree with @MehmedGIT here, While I tend to use the `from .. import ... as ...` pattern a lot, it should be optional and clear, concise, unambiguous class names are preferable.

I find non-qualified `Job` more cryptic than a longer class name that makes it clear what this class is for.

Reply:
@kba, I still think the `PY` prefix is not ideal for the Pydantic input/output models used for interaction with the Processing Server. Any better suggestions?

Reply:
maybe a `Model` or `PydanticModel` suffix? But TBH, I can live with `PY` prefix just fine.

Reply:
Let's keep it then PY for now.